worlds/banjo_tooie/__init__.py

```diff
@@ -589,114 +585,112 @@ def generate_early(self) -> None:
             raise OptionError(
                 "You cannot have progressive Beak Buster without randomizing moves and randomizing BK moves"
             )
-        if (self.options.egg_behaviour == EggsBehaviour.option_random_starting_egg
-                or self.options.egg_behaviour == EggsBehaviour.option_simple_random_starting_egg) \
-                and (not self.options.randomize_bk_moves or not self.options.randomize_bt_moves):
+        if (self.options.egg_behaviour.value == EggsBehaviour.option_random_starting_egg
+                or self.options.egg_behaviour.value == EggsBehaviour.option_simple_random_starting_egg) \
+                and (not self.options.randomize_bk_moves.value or not self.options.randomize_bt_moves.value):
             raise OptionError(
                 "You cannot have Randomize Starting Egg without randomizing moves and randomizing BK moves"
             )
-        elif self.options.egg_behaviour == EggsBehaviour.option_progressive_eggs\
-                and not self.options.randomize_bt_moves:
+        elif self.options.egg_behaviour.value == EggsBehaviour.option_progressive_eggs\
+                and not self.options.randomize_bt_moves.value:
             raise OptionError("You cannot have progressive Eggs without randomizing moves")
-        if self.options.progressive_shoes\
+        if self.options.progressive_shoes.value\
                 and not (
-                    self.options.randomize_bk_moves
-                    and self.options.randomize_bt_moves
-                    and (self.options.randomize_signposts or self.options.nestsanity)
+                    self.options.randomize_bk_moves.value
+                    and self.options.randomize_bt_moves.value
+                    and (self.options.randomize_signposts.value or self.options.nestsanity.value)
                 ):
-            raise OptionError("You cannot have progressive Shoes without randomizing moves,\
-                randomizing BK moves and enabling either nestanity or randomize signpost")
-        if self.options.progressive_water_training != ProgressiveWaterTraining.option_none \
+            raise OptionError("You cannot have progressive Shoes without randomizing moves, "
+                              "randomizing BK moves and enabling either nestanity or randomize signpost")
+        if self.options.progressive_water_training.value != ProgressiveWaterTraining.option_none \
                 and (
-                    self.options.randomize_bk_moves == RandomizeBKMoveList.option_none
-                    or not self.options.randomize_bt_moves
+                    self.options.randomize_bk_moves.value == RandomizeBKMoveList.option_none
+                    or not self.options.randomize_bt_moves.value
                 ):
             raise OptionError("You cannot have progressive Water Training\
                 without randomizing moves and randomizing BK moves")
```

Comment:
```suggestion
            raise OptionError("You cannot have progressive Water Training "
                              "without randomizing moves and randomizing BK moves")
```
Same thing as this comment. https://github.com/ArchipelagoMW/Archipelago/pull/4471#discussion_r2557928479